### PR TITLE
Support for analytics plugin

### PR DESCRIPTION
**Pull request by roborourke**
- [x] Add documentation
- [x] Add config for AB Test feature

### CODE REVIEW

**Comment by roborourke** [comment]
Need to merge https://github.com/humanmade/altis-core/pull/44 after this one is merged.

File: docs/native.md [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+				"analytics": {
+					"native": false
+				}
+			}
+		}
+	}
+}
+```
+
+Analytics data is stored in an S3 data lake and delivered to Elasticsearch where it is is 1-5 minutes behind real-time.
```
Comment (joehoyle) [inline docs/native.md]:
```suggestion
Analytics data is stored in an S3 data lake and delivered to Elasticsearch where it is 1-5 minutes behind real-time.
```

File: docs/native.md [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+
+The data is queried from Elasticsearch in the application providing a powerful query language with filtering and aggregations for collecting statistics and other metrics.
+
+This feature provides a tool for developers to query the data for any use-case imaginable including personalisation and [AB testing](./ab-testing.md).
+
+Look out for new built in features backed by native analytics coming in future versions including [Experience Blocks](https://www.altis-dxp.com/experience-blocks/).
```
Comment (joehoyle) [inline docs/native.md]: IMO the dev docs isn't the place for this.

File: docs/native.md [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+
+## Usage
+
+### In JavaScript
+
+Once installed the plugin will queue up an analytics tracker script that provides a few client side functions you can use:
```
Comment (joehoyle) [inline docs/native.md]: I don't think we need "once installed" here.

File: docs/native.md [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+
+Get the current time since the unix epoch in milliseconds.
+
+**`Altis\Analytics\Utils\merge_aggregations( array $current, array $new, string $bucket_type = '' ) : array`**
+
+A utility function for merging aggregations returned by Elasticsearch queries. This is necessary to keep a store of your analytics data as Elasticsearch indexes can be rotated or lost.
+
+## Querying Data
+
+Data can be queried from Elasticsearch, providing a powerful tool for filtering and aggregating records. If you are using Altis requests are automatically signed.
```
Comment (joehoyle) [inline docs/native.md]: This plugin / Altis docs is a bit confusing. Can we tweak all these docs to be specific to use in Altis?

File: docs/native.md [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+  - `utm_source`: The Urchin Tracker source from the query string if set.
+  - `utm_medium`: The Urchin Tracker medium from the query string if set.
+  - Any attributes added via the `altis.analytics.data.attributes` filter.
+  - Any attributes added via `Altis.Analytics.registerAttribute()` or passed to `Altis.Analytics.record()`.
+- `metrics`: Numeric data associated with an event.
+  - `scrollDepthMax`: Maximum scroll depth on page so far. Percentage value between 1-100.
+  - `scrollDepthNow`: Scroll depth at time of event. Percentage value between 1-100.
+  - `elapsed`: Time elapsed in milliseconds since the start of the page view.
+  - Any metrics added via `Altis.Analytics.registerMetric()` or passed to `Altis.Analytics.record()`.
+- `endpoint`
```
Comment (joehoyle) [inline docs/native.md]: As I understand it, and `endpoint` is unique per user, but what is a "user". If I log in to my phone and computer, do I use the same endpoint? If I am logged out, and then log in, is that using the same endpoint?

If I update values `enpdoint.Demographic.Locale`, is that just setting those value for this event, or will it also set the endpoint ("user") data for subsequent records (i.e. the endpoint/user data is a separate, single object?)

Comment (roborourke) [reply docs/native.md]: Soz - an endpoint is a device & browser. They're only connected via `endpoint.User.UserId` if set, eg. if you log in on multiple browsers / devices

File: docs/native.md (hunk 4 above)
Comment (roborourke) [reply docs/native.md]: My bad, copy pasta fail. I'll go through them again and double check

File: docs/native.md (hunk 5 above)
Comment (roborourke) [reply docs/native.md]: Oh and yeah - whatever data is associated with the endpoint will be automatically merged in with each event record, so they're handled separately.

Comment (roborourke) [reply docs/native.md]: I added a new "Key Concepts" section that answers your questions 👍

**Pull request merged**